### ISSUE TITLE
1.1.0 Release

### ISSUE THREAD

**Issue by bkchr** [polkadot-fellows/runtimes#68]
Let's already start with the list of things to include for the 1.1.0 release. Please everyone leave a comment with a link to a pr or issue, so we have it tracked.

I would propose that we directly start with the release after we have finally finished `1.0.0`.

**Comment by bkontur** [comment]
https://github.com/polkadot-fellows/runtimes/pull/45
https://github.com/polkadot-fellows/runtimes/pull/48

**Comment by joepetrowski** [comment]
- [ ] https://github.com/polkadot-fellows/runtimes/pull/62
- [ ] https://github.com/polkadot-fellows/runtimes/pull/54
- [ ] https://github.com/polkadot-fellows/runtimes/pull/65

**Comment by acatangiu** [comment]
- [ ] https://github.com/polkadot-fellows/runtimes/pull/65

**Comment by bkchr** [comment]
I would like that we slowly start with the release. I think the best would be to get @muharem's PR merged first: https://github.com/polkadot-fellows/runtimes/pull/56

Probably the best to keep this pr and then create another one that updates to 1.3.0.

**Comment by franciscoaguirre** [comment]
When that PR is merged, I can make one updating to 1.3.0

**Comment by acatangiu** [comment]
> When that PR is merged, I can make one updating to 1.3.0

afaik @svyatonik is already working on this: https://github.com/svyatonik/runtimes/pull/2

**Comment by muharem** [comment]
it would be nice to get this PR to the release https://github.com/polkadot-fellows/runtimes/pull/84
I can not base it on https://github.com/polkadot-fellows/runtimes/pull/56 since the branch in my fork

**Comment by bkchr** [comment]
> will we include polkadot-sdk#1.3 to the this release? @bkchr

As I already said above, I would do it.

**Comment by muharem** [comment]
one more candidate, simple migration removal https://github.com/polkadot-fellows/runtimes/pull/54

**Comment by Sophia-Gold** [comment]
- [ ] https://github.com/polkadot-fellows/runtimes/pull/90

**Comment by muharem** [comment]
- [ ] [must have] https://github.com/polkadot-fellows/runtimes/pull/107
- [ ] [very nice to have] https://github.com/polkadot-fellows/runtimes/pull/109

**Comment by bkchr** [comment]
So, all the prs mentioned above are merged. I still want to include https://github.com/polkadot-fellows/runtimes/pull/117 and https://github.com/polkadot-fellows/runtimes/pull/118. Maybe also @Leemo94 also wants to bring up his small pr for changing the Kusama OpenGov tracks config. However, I would not wait on this if the pr is not popping up today.

Otherwise we still require a weights update. @ggwpez @liamaharon can you please do this?

**Comment by NachoPal** [comment]
About the weights, I found that [this](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/asset-hubs/asset-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#L100C1-L106C3) will provoke an Overweight message.

**Comment by acatangiu** [comment]
FYI: many (if not all) of the required benchmarks have already been run against (relatively) latest `main` in https://github.com/polkadot-fellows/runtimes/pull/108 - you could also get them from there

**Comment by NachoPal** [comment]
Since the introduction of https://github.com/paritytech/polkadot-sdk/pull/1234, the [teleport](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/collectives/collectives-polkadot/src/impls.rs#L66) when kicking a member will fail if the `AlliancePalletAccount` does not have some extra balance to pay the fees.

`AlliancePalletAccount` (`13UVJyLnbVp6r4JgFxb88FAvQBL5WCdE2FU8uzyCJCHkDhx1`) should be funded to pay the fees or otherwise it will fail in production.

**Comment by joepetrowski** [comment]
We are planning to just add an `UnpaidTeleport` instruction to XCMv4.

**Comment by bkontur** [comment]
> About the weights, I found that [this](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/asset-hubs/asset-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#L100C1-L106C3) will provoke an Overweight message.

@NachoPal Do you have any specific scenario in mind?


Actually, relay chains and system parachains do not recognize any trusted reserves `type IsReserve = ();`, so they cannot process `ReserveAssetDeposited`. When we regenerate new benchmarks, then [this](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/asset-hubs/asset-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#L100C1-L106C3) `500_000_000_000` will become `18_446_744_073_709_551_000` (as a [`Weight::MAX`](https://github.com/paritytech/polkadot-sdk/pull/1726)), this is ok since Gav removed remote weight estimation [here](https://github.com/paritytech/polkadot-sdk/pull/1645/files#diff-742026ed87c0710683d2ef4263b463a3e6f0acfae3c8fad80f2f3d291088997fL1221-L1236). If you want to process `ReserveAssetDeposited`, you need to set up correct benchmark and `IsReserve`, e.g. as we did for bridges [here](https://github.com/polkadot-fellows/runtimes/pull/108/files#diff-ab055e2290ada4126867ec11381ab9f3993651dfac296636e09cea7b90d71746L98-R90).

**Comment by acatangiu** [comment]
What about https://github.com/polkadot-fellows/runtimes/pull/108? It's reviewed and ready to merge IMO and only _adds_ functionality that needs further governance actions to actually enable.

It'd be great to get it in this upcoming release!

**Comment by muharem** [comment]
we might want to include this small PR, one constant value change, otherwise the current salary budget limit, might not be enough 
- [ ] https://github.com/polkadot-fellows/runtimes/pull/121

**Comment by NachoPal** [comment]
`StakingAdmin`, `FellowshipAdmin` Origins should be waived from paying fees, otherwise it will fail when trying to send xcm messages.

**Comment by mustermeiszer** [comment]
@bkchr will this also include https://github.com/polkadot-fellows/runtimes/pull/67 automatically?

**Comment by joepetrowski** [comment]
Yes

**Comment by muharem** [comment]
> Since the introduction of [paritytech/polkadot-sdk#1234](https://github.com/paritytech/polkadot-sdk/pull/1234), the [teleport](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/collectives/collectives-polkadot/src/impls.rs#L66) when kicking a member will fail if the `AlliancePalletAccount` does not have some extra balance to pay the fees.
> 
> `AlliancePalletAccount` (`13UVJyLnbVp6r4JgFxb88FAvQBL5WCdE2FU8uzyCJCHkDhx1`) should be funded to pay the fees or otherwise it will fail in production.
> 
> Similarly will happen with `ReferendaPalletAccount` (`13UVJyLnbVp9974YNNtrNWNjJQhBYSnMoTQycaARVAt49q7h`).

fixed with this PR - https://github.com/polkadot-fellows/runtimes/pull/125

**Comment by NachoPal** [comment]
> StakingAdmin, FellowshipAdmin, etc.. Origins should be waived from paying fees, otherwise it will fail when trying to send xcm messages.

This was fixed also in #125

**Comment by bkchr** [comment]
The following prs are still missing:

- [ ] #125 
- [ ] #129
- [ ] #119
- [ ] #108

**Comment by ggwpez** [comment]
All ticked off now, can we go ahead?

**Comment by joepetrowski** [comment]
> All ticked off now, can we go ahead?

@NachoPal was doing a final run of integration tests. Any ETA on them finishing?

**Comment by NachoPal** [comment]
> @NachoPal was doing a final run of integration tests. Any ETA on them finishing?

All green, but System Parachains `spec_version` still need to be bumped.

**Comment by joepetrowski** [comment]
https://github.com/polkadot-fellows/runtimes/pull/138

**Comment by NachoPal** [comment]
> > About the weights, I found that [this](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/asset-hubs/asset-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#L100C1-L106C3) will provoke an Overweight message.
> 
> @NachoPal Do you have any specific scenario in mind?
> 
> Actually, relay chains and system parachains do not recognize any trusted reserves `type IsReserve = ();`, so they cannot process `ReserveAssetDeposited`. When we regenerate new benchmarks, then [this](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/asset-hubs/asset-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#L100C1-L106C3) `500_000_000_000` will become `18_446_744_073_709_551_000` (as a [`Weight::MAX`](https://github.com/paritytech/polkadot-sdk/pull/1726)), this is ok since Gav removed remote weight estimation [here](https://github.com/paritytech/polkadot-sdk/pull/1645/files#diff-742026ed87c0710683d2ef4263b463a3e6f0acfae3c8fad80f2f3d291088997fL1221-L1236). If you want to process `ReserveAssetDeposited`, you need to set up correct benchmark and `IsReserve`, e.g. as we did for bridges [here](https://github.com/polkadot-fellows/runtimes/pull/108/files#diff-ab055e2290ada4126867ec11381ab9f3993651dfac296636e09cea7b90d71746L98-R90).

Yes, you are right that that message should fail, but it should fail with a `UntrustedReserveLocation` and not with an Overweight. It is properly working now.

**Comment by bkontur** [comment]
> > > About the weights, I found that [this](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/asset-hubs/asset-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#L100C1-L106C3) will provoke an Overweight message.
> > 
> > 
> > @NachoPal Do you have any specific scenario in mind?
> > Actually, relay chains and system parachains do not recognize any trusted reserves `type IsReserve = ();`, so they cannot process `ReserveAssetDeposited`. When we regenerate new benchmarks, then [this](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/asset-hubs/asset-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#L100C1-L106C3) `500_000_000_000` will become `18_446_744_073_709_551_000` (as a [`Weight::MAX`](https://github.com/paritytech/polkadot-sdk/pull/1726)), this is ok since Gav removed remote weight estimation [here](https://github.com/paritytech/polkadot-sdk/pull/1645/files#diff-742026ed87c0710683d2ef4263b463a3e6f0acfae3c8fad80f2f3d291088997fL1221-L1236). If you want to process `ReserveAssetDeposited`, you need to set up correct benchmark and `IsReserve`, e.g. as we did for bridges [here](https://github.com/polkadot-fellows/runtimes/pull/108/files#diff-ab055e2290ada4126867ec11381ab9f3993651dfac296636e09cea7b90d71746L98-R90).
> 
> Yes, you are right, that message should fail, but it should fail with an `UntrustedReserveLocation` and not with an Overweight. It is properly working now.

we had a similar discussions with @acatangiu around here [here](https://github.com/polkadot-fellows/runtimes/pull/129#issuecomment-1878631594)

hmm, maybe, if we know that `IsReserve = ()` is no-op, than the benchmark should set `Weight::zero()` instead of `T::BlockWeights::get().max_block` or `Weight::MAX`.
But it could be risky, e.g. if we change `type IsReseve = ...` and forget to regenerate benchmarks...

I don't know, what is the best solution here. Regardless of returned xcm error (`WeightNotComputable` vs. `UntrustedReserveLocation`), `Weight::MAX` seems safer.

**Comment by bkchr** [comment]
Going to close the issue, as the release is finished.
